test/common/upstream/upstream_impl_test.cc

```diff
@@ -2070,6 +2070,25 @@ TEST(HostsPerLocalityImpl, Filter) {
   }
 }
 
+TEST(HostSetImplTest, UpdateCallbacks) {
```

Comment:
I was hoping for  a test where you have both callback types, and then do a series of updates that should impact one or both callbacks and check that. Is that possible?

Reply:
Ah yeah i misunderstood. 

Since this PR doesn't change when the callbacks are called, just splits them up, we won't see them executed at different times. I wanted to split up the API change and the behavioral change.

 That said I can add a test that shows when the callbacks are invoked currently to make it easier to see how that changes in subsequent PRs.

Reply:
OK makes sense.